src/main/java/uk/gov/justice/laa/crime/cfecrime/utils/RequestHandler.java

```diff
@@ -1,45 +1,121 @@
     }
+
+    public void setFullMeansTestOutcome(StatelessApiResponse statelessApiResponse, CfeCrimeRequest cfeCrimeRequest, CfeCrimeResponse cfeCrimeResponse) {
+
+        Outcome fullOutcome = null;
+        CaseType caseType = null;
+        MagCourtOutcome magCourtOutcome = null;
+        if (statelessApiResponse.getFullMeansAssessment().getResult() != null) {
+
+            caseType = cfeCrimeRequest.getSectionInitialMeansTest().getCaseType();
+            magCourtOutcome = cfeCrimeRequest.getSectionInitialMeansTest().getMagistrateCourtOutcome();
+
+            FullAssessmentResult statelessfullResult = statelessApiResponse.getFullMeansAssessment().getResult();
+            fullOutcome = FullMeansTestOutcomeCalculator.getFullMeansTestOutcome(statelessfullResult, caseType, magCourtOutcome);
+            cfeCrimeResponse.setOutcome(fullOutcome);
+        }
+        cfeCrimeResponse.setOutcome(fullOutcome);
+
+    }
+
+    public void setInitialMeansTestOutcome(StatelessApiResponse statelessApiResponse, CfeCrimeRequest cfeCrimeRequest, CfeCrimeResponse cfeCrimeResponse) throws UndefinedOutcomeException {
+        CaseType caseType = null;
+        MagCourtOutcome magCourtOutcome = null;
+
+        Outcome initOutcome = null;
+        if (cfeCrimeRequest.getSectionInitialMeansTest() != null) {
+            caseType = cfeCrimeRequest.getSectionInitialMeansTest().getCaseType();
+            if (cfeCrimeRequest.getSectionInitialMeansTest().getMagistrateCourtOutcome() != null) {
+                magCourtOutcome = cfeCrimeRequest.getSectionInitialMeansTest().getMagistrateCourtOutcome();
+            }
+            InitAssessmentResult initAssessmentResult = statelessApiResponse.getInitialMeansAssessment().getResult();
+            initOutcome = InitMeansTestOutcomeCalculator.getInitMeansTestOutcome(initAssessmentResult, statelessApiResponse.getInitialMeansAssessment().isFullAssessmentPossible());
+        }
+        cfeCrimeResponse.setOutcome(initOutcome);
+
+    }
+
+    private static void buildCmaRequest(CfeCrimeRequest cfeCrimeRequest, StatelessApiRequest statelessApiRequest){
+
+        Assessment assessment = new Assessment();
+        if (cfeCrimeRequest.getAssessment() != null) {
+            assessment.setAssessmentType(cfeCrimeRequest.getAssessment().getAssessmentType());
+            assessment.setAssessmentDate(cfeCrimeRequest.getAssessment().getAssessmentDate());
+            statelessApiRequest.setAssessment(assessment);
+        }
+
+        if (cfeCrimeRequest.getSectionInitialMeansTest() != null && cfeCrimeRequest.getSectionInitialMeansTest().getIncome() != null) {
+            statelessApiRequest.setIncome(cfeCrimeRequest.getSectionInitialMeansTest().getIncome());
+        }
+        if (cfeCrimeRequest.getSectionFullMeansTest() != null && cfeCrimeRequest.getSectionFullMeansTest().getOutgoings() != null) {
+            statelessApiRequest.setOutgoings(cfeCrimeRequest.getSectionFullMeansTest().getOutgoings());
+        }
+
```

Comment:
Looks like you have forgotten the MagCourtOutcome and the CaseType? (and maybe others?)

Reply:
It is because it is not showing as needing setting up. I will look into why that is.